scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -108,6 +108,8 @@ trait JavaModule extends mill.Module with TaskModule { outer =>
     }
   }
 
+  def assemblyRules: T[Seq[Assembly.Rule]] = T { Assembly.defaultRules }
```

Comment:
Do we need to keep it as Target, or we can go with plain def here? Cause if we can go with def, we can remove serialization code for Assembly.Rule

Reply:
I don't have any real preference. If we go with def, we could also include rules that are not serializable, e.g. containing lambdas. I leave that up to you

Reply:
If we make it def, instead of `T`, and the users change them while working on their build, will it reevaluate for sure? I tend to think that it will, because we invalidate caches on build change